### ISSUE TITLE
How should an invalid fieldOrder value be handled?

### ISSUE THREAD

**Issue by nbarbettini** [stormpath/stormpath-framework-spec#63]
If my configuration for some form route looks like this:

``` yaml
form: 
  fields: 
    login: 
      [...]
    password: 
      [...]
  fieldOrder: 
    - login
    - passwd
```

What should happen when the incorrect/missing value `passwd` is encountered? Should an error be thrown (and when)?

**Comment by lhazlewood** [comment]
IMO an error should always be surfaced as soon as it is found, i.e. 'fail fast' to let the user know that configuration is invalid.  This ensures that it is always clean - the alternative is a polluted config file over time.

**Comment by nbarbettini** [comment]
Makes sense. I'm curious if we should spec it to fail at startup, or the first time said route is accessed. I'd be in favor of the former, since that gives the developer more feedback.

**Comment by lhazlewood** [comment]
I prefer the former as well - the earlier in the app lifecycle the better IMO.

**Comment by robertjd** [comment]
Yes, we should always fail at startup for configuration issues.  Very annoying otherwise.  FYI, we deemed field order configuration validation to be low prio for Express, we just don't show the field if it's not in the fields hash.

**Comment by nbarbettini** [comment]
Alright, sounds like we're going in the same direction here. I think this should be in the spec - expanding [configuration.md](https://github.com/stormpath/stormpath-framework-spec/blob/master/configuration.md) to include additional validation concerns.

Whether this should be in v1.0 or v1.1, I don't know. As @robertjd said, it's fairly low prio, but I think it should be in there at some point.

**Comment by bretterer** [comment]
@nbarbettini @robertjd My vote is v1.1 for this

**Comment by nbarbettini** [comment]
Just realized that #67 is the same thing, just a wider scope.

**Comment by edjiang** [comment]
Closed in favor of #67
